pkg/admission/validator/shoot.go

```diff
@@ -31,24 +32,15 @@ type shoot struct {
 }
 
 // NewShootValidator returns a new instance of a shoot validator.
-func NewShootValidator() extensionswebhook.Validator {
-	return &shoot{}
-}
-
-// InjectScheme injects the given scheme into the validator.
-func (s *shoot) InjectScheme(scheme *runtime.Scheme) error {
-	s.decoder = serializer.NewCodecFactory(scheme, serializer.EnableStrict).UniversalDecoder()
-	return nil
-}
-
-// InjectClient injects the given client into the validator.
-func (s *shoot) InjectClient(client client.Client) error {
-	s.client = client
-	return nil
+func NewShootValidator(mgr manager.Manager) extensionswebhook.Validator {
```

Comment:
I would rather prefer to accept only the needed things instead of accepting a general `manager.Manager`. See https://github.com/gardener/gardener-extension-registry-cache/blob/5fdb7552139cc3a4ee98e460270b658ee0e188ab/pkg/admission/validator/shoot.go#L38-L42.

```suggestion
func NewShootValidator(client client.Client, decoder runtime.Decoder) extensionswebhook.Validator {
```